src/components/Menus/DropdownMenu.js

```diff
@@ -0,0 +1,207 @@
+    if (nextProps.open !== null && this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.open()
+      } else {
+        this.close()
+      }
+    }
+  }
+
+  handleSelect = (option) => {
+    const { onSelect, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(false)
+    } else {
+      // if multi then do not close
+      this.close()
+    }
+
+    onSelect && onSelect(option)
+  }
+
+  handleClick = (event) => {
+    event.preventDefault()
+    const { open, onClick, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(!open)
+      return
+    }
+
+    if (!this.state.open) {
+      this.open()
+    } else {
+      this.close()
+    }
+    onClick && onClick(event)
+  }
+
+  handleMenuBlur = () => {
+    const { onRequestChange } = this.props
+    // Timeout ensures click event has correct state when attempting to close
+    setTimeout( () => {
+      if (this.controlledOpen()) {
+        onRequestChange && onRequestChange(false)
+      } else {
+        this.close()
+      }
+    }, 100)
+  }
+
+  handleKeyDown = (event) => {
+    const { onRequestChange } = this.props
+    switch(event.key) {
+      case 'Enter':
+        if (!event.defaultPrevented && this.controlledOpen()) {
+          event.preventDefault()
+          onRequestChange && onRequestChange(true)
+        } else {
+          event.preventDefault()
+          if (this.state.open) {
+            this.close()
+          } else {
+            this.open()
+          }
+        }
+        break
+      case 'Escape':
+        event.preventDefault()
+
+        if (this.controlledOpen()) {
+          onRequestChange && onRequestChange(false)
+        } else {
+          this.state.open && this.close()
+        }
+        break
+    }
+  }
+
+  controlledOpen() {
+    return this.props.open !== null
+  }
+
+  open() {
+    const { onOpen } = this.props
+
+    this.setState({open: true}, () => {
+      this.menu.focus()
+      onOpen && onOpen()
+    })
+  }
+
+  close() {
+    const { onClose } = this.props
+    this.setState({open: false}, () => {
+      this.menu.blur()
+      onClose && onClose()
+    })
+  }
+
+  renderTriggerElement() {
+    const { triggerElement } = this.props
+    const { open } = this.state
+    if (triggerElement) {
+      return React.cloneElement(triggerElement, {
+        onClick: this.handleClick,
+        'aria-haspopup': true,
+        'aria-expanded': open,
+      })
+    }
+  }
+
+  render() {
+    const {
+      style,
+      children,
+      menuProps
+    } = this.props
+
+    const isOpen = this.state.open
+
+    return (
+      <div
+        onKeyDown={this.handleKeyDown}
+      >
+        <div style={{position: 'relative'}}>
+          { this.renderTriggerElement() }
+
+          <div
+            style={[
+              styles.menuContainer,
+              style,
```

Comment:
I would expect a passed down `style` prop to apply to the top level html element. I'm imaging a case like:
```
<div style={{ display: flex }} >
  <div>Logo</div>
  <DropdownMenu style={{ flexGrow: 1 }} />
</div>
```

The above wouldn't render as I would expect and would be impossible to implement.